# Initialization

In [1]:
import polars as pl

from adex.helpers import load_data_per_condition, gene_intersection
from adex.models import Condition, ConditionSequencingTissueDataLoader, SequencingTechnique, TissueEnum
from adex.helpers import high_frequency_genes_dataframe, common_genes_dataframe, get_pre_processed_dataset
from adex.models import ConditionDataLoader

files_path = "../data/adex-database/samples"
metadata_path = "../data/adex-database/metadata.csv"
datasets_info_path = "../data/adex-database/datasets_info.csv"

ra_data  = load_data_per_condition(Condition.RA, files_path)
t1d_data = load_data_per_condition(Condition.T1D, files_path)
ssc_data = load_data_per_condition(Condition.SSc, files_path)
sle_data = load_data_per_condition(Condition.SLE, files_path)
sjs_data = load_data_per_condition(Condition.SjS, files_path)

# Explore Metadata

In [2]:
metadata_df = pl.read_csv("../data/adex-database/metadata.csv")
metadata_df.shape

(3980, 10)

In [3]:
metadata_df

shape: (3_980, 10)
┌────────────┬──────────┬──────────────┬────────┬───┬───────────┬────────┬──────┬───────────┐
│ Sample     ┆ GSE      ┆ Experimental ┆ GPL    ┆ … ┆ Cell Type ┆ Gender ┆ Age  ┆ Ethnicity │
│ ---        ┆ ---      ┆ Strategy     ┆ ---    ┆   ┆ ---       ┆ ---    ┆ ---  ┆ ---       │
│ str        ┆ str      ┆ ---          ┆ str    ┆   ┆ str       ┆ str    ┆ str  ┆ str       │
│            ┆          ┆ str          ┆        ┆   ┆           ┆        ┆      ┆           │
╞════════════╪══════════╪══════════════╪════════╪═══╪═══════════╪════════╪══════╪═══════════╡
│ GSM260887  ┆ GSE10325 ┆ Expression   ┆ GPL96  ┆ … ┆ B cells   ┆ null   ┆ null ┆ null      │
│ GSM260890  ┆ GSE10325 ┆ Expression   ┆ GPL96  ┆ … ┆ B cells   ┆ null   ┆ null ┆ null      │
│ GSM260892  ┆ GSE10325 ┆ Expression   ┆ GPL96  ┆ … ┆ B cells   ┆ null   ┆ null ┆ null      │
│ GSM260895  ┆ GSE10325 ┆ Expression   ┆ GPL96  ┆ … ┆ B cells   ┆ null   ┆ null ┆ null      │
│ GSM260898  ┆ GSE10325 ┆ Expression   ┆ GPL96  ┆ … ┆ B cells   ┆ null   ┆ null ┆ null      │
│ …          ┆ …        ┆ …            ┆ …      ┆ … ┆ …         ┆ …      ┆ …    ┆ …         │
│ GSM2495622 ┆ GSE95065 ┆ Expression   ┆ GPL570 ┆ … ┆ null      ┆ null   ┆ null ┆ null      │
│ GSM2495623 ┆ GSE95065 ┆ Expression   ┆ GPL570 ┆ … ┆ null      ┆ null   ┆ null ┆ null      │
│ GSM2495624 ┆ GSE95065 ┆ Expression   ┆ GPL570 ┆ … ┆ null      ┆ null   ┆ null ┆ null      │
│ GSM2495625 ┆ GSE95065 ┆ Expression   ┆ GPL570 ┆ … ┆ null      ┆ null   ┆ null ┆ null      │
│ GSM2495626 ┆ GSE95065 ┆ Expression   ┆ GPL570 ┆ … ┆ null      ┆ null   ┆ null ┆ null      │
└────────────┴──────────┴──────────────┴────────┴───┴───────────┴────────┴──────┴───────────┘

In [4]:
total_samples = metadata_df.select("Sample").shape[0]
unique_samples = metadata_df.select("Sample").unique().shape[0]

conditions_df = metadata_df.select("Condition").unique()
conditions_count = conditions_df.shape[0]
conditions = conditions_df.to_series().to_list()

tissues_df = metadata_df.select("Tissue").unique()
tissues_count = tissues_df.shape[0]
tissues = tissues_df.to_series().to_list()

gpl_df = metadata_df.select("GPL").unique()
gpl_count = gpl_df.shape[0]
gpls = gpl_df.to_series().to_list()

gse_df = metadata_df.select("GSE").unique()
gse_count = gse_df.shape[0]
gses = gse_df.to_series().to_list()

males = metadata_df.select("Gender").filter(pl.col("Gender") == "Male").shape[0]
females = metadata_df.select("Gender").filter(pl.col("Gender") == "Female").shape[0]
gender_unknown = metadata_df.select("Gender").filter((pl.col("Gender") != "Female") & (pl.col("Gender") != "Male")).shape[0]

strategy_df = metadata_df.select("Experimental Strategy").unique()

print(f"""
Total Samples = {total_samples}
Unique Samples = {unique_samples}
Conditions = {conditions_count} [{",".join(conditions)}]
GPL = {gpl_count} [{",".join(gpls)}]
GSE = {gse_count} [{",".join(gses)}]
Tissues = {tissues_count} [{",".join(tissues)}]
Males = {males}, Females = {females}, Unknown Gender = {gender_unknown}
Experimental Strategy = {strategy_df.to_series().to_list()}
""")


Total Samples = 3980
Unique Samples = 3854
Conditions = 6 [RA,SjS,T1D,SLE,SSc,Healthy]
GPL = 13 [GPL97,GPL14951_GPL13534,GPL13158,GPL11154,GPL96,GPL20301,GPL6884,GPL570,GPL15456,GPL10558_GPL13534,GPL10558,GPL13667,GPL16791]
GSE = 41 [GSE89408,GSE56649,GSE51092,GSE110174,GSE55098,GSE124073,GSE55235,GSE7451,GSE50772,GSE13887,GSE63903,GSE77298,GSE65010,GSE12021,GSE11907,GSE40611,GSE38351,GSE60424,GSE10325,GSE82221,GSE72509,GSE80183,GSE24706,GSE65391,GSE110914,GSE23117,GSE55457,GSE90081,GSE110169,GSE45291,GSE117931,GSE112341,GSE108497,GSE30153,GSE124939,GSE104174,GSE57383,GSE61635,GSE93683,GSE95065,GSE84844]
Tissues = 7 [Parotid gland,Peripheral blood,Salivary gland,Skin,Saliva,Whole blood,Synovial membrane]
Males = 269, Females = 1547, Unknown Gender = 0
Experimental Strategy = ['Expression', 'Expression + Methylation']


In [5]:
metadata_df.group_by("Condition").agg(pl.struct(["Sample"]).n_unique().alias('SamplesPerCondition')).sort("SamplesPerCondition", descending=True)

shape: (6, 2)
┌───────────┬─────────────────────┐
│ Condition ┆ SamplesPerCondition │
│ ---       ┆ ---                 │
│ str       ┆ u32                 │
╞═══════════╪═════════════════════╡
│ SLE       ┆ 1502                │
│ Healthy   ┆ 957                 │
│ RA        ┆ 862                 │
│ SjS       ┆ 281                 │
│ SSc       ┆ 157                 │
│ T1D       ┆ 95                  │
└───────────┴─────────────────────┘

In [6]:
metadata_df.group_by("Age").agg(pl.struct(["Sample"]).n_unique().alias('SamplesPerAge')).sort("Age")


shape: (10, 2)
┌───────┬───────────────┐
│ Age   ┆ SamplesPerAge │
│ ---   ┆ ---           │
│ str   ┆ u32           │
╞═══════╪═══════════════╡
│ null  ┆ 2400          │
│ 1-10  ┆ 113           │
│ 11-20 ┆ 342           │
│ 21-30 ┆ 383           │
│ 31-40 ┆ 331           │
│ 41-50 ┆ 105           │
│ 51-60 ┆ 66            │
│ 61-70 ┆ 69            │
│ 71-80 ┆ 39            │
│ >80   ┆ 6             │
└───────┴───────────────┘

# Check Metadata for duplicates

Duplicates will create issues later in the process if there is a join with this file. 

In [7]:
metadata_df.group_by("Sample").len().filter(pl.col("len") > 1).sort("len", descending=True).head()

shape: (5, 2)
┌────────────┬─────┐
│ Sample     ┆ len │
│ ---        ┆ --- │
│ str        ┆ u32 │
╞════════════╪═════╡
│ GSM940478  ┆ 2   │
│ GSM2981155 ┆ 2   │
│ GSM2981203 ┆ 2   │
│ GSM940473  ┆ 2   │
│ GSM2981058 ┆ 2   │
└────────────┴─────┘

In [9]:
# Check removing duplicates 
metadata_df.unique(subset=["Sample"]).group_by("Sample").len().sort("len", descending=True).head()

shape: (5, 2)
┌────────────┬─────┐
│ Sample     ┆ len │
│ ---        ┆ --- │
│ str        ┆ u32 │
╞════════════╪═════╡
│ GSM2902068 ┆ 1   │
│ GSM1863695 ┆ 1   │
│ GSM1101281 ┆ 1   │
│ GSM2902264 ┆ 1   │
│ GSM301752  ┆ 1   │
└────────────┴─────┘

# Explore Files that seem problematic

### GSE95065

There seem to be two files that need to be merged into one 


In [7]:
GSE95065_large = pl.read_parquet("../data/adex-database/samples/archived_not_used/GSE95065.tsv.parquet")
GSE95065_large

shape: (10_477, 34)
┌─────────┬────────────┬───────────┬───────────┬───┬───────────┬───────────┬───────────┬───────────┐
│ gene    ┆ GSM2495594 ┆ GSM249559 ┆ GSM249559 ┆ … ┆ GSM249562 ┆ GSM249562 ┆ GSM249562 ┆ GSM249562 │
│ ---     ┆ ---        ┆ 5         ┆ 6         ┆   ┆ 3         ┆ 4         ┆ 5         ┆ 6         │
│ str     ┆ f64        ┆ ---       ┆ ---       ┆   ┆ ---       ┆ ---       ┆ ---       ┆ ---       │
│         ┆            ┆ f64       ┆ f64       ┆   ┆ f64       ┆ f64       ┆ f64       ┆ f64       │
╞═════════╪════════════╪═══════════╪═══════════╪═══╪═══════════╪═══════════╪═══════════╪═══════════╡
│ RFC2    ┆ 7.472319   ┆ 7.520892  ┆ 7.46077   ┆ … ┆ 7.379748  ┆ 7.278312  ┆ 7.269157  ┆ 7.1024    │
│ HSPA6   ┆ 6.655682   ┆ 6.248444  ┆ 6.563427  ┆ … ┆ 6.119981  ┆ 6.749708  ┆ 6.33245   ┆ 6.380119  │
│ PAX8    ┆ 6.772054   ┆ 6.926438  ┆ 7.034588  ┆ … ┆ 7.145639  ┆ 7.119249  ┆ 7.113005  ┆ 6.452059  │
│ THRA    ┆ 6.90694    ┆ 6.48762   ┆ 6.229516  ┆ … ┆ 5.903811  ┆ 6.093964  ┆ 5.95448   ┆ 6.172006  │
│ …       ┆ …          ┆ …         ┆ …         ┆ … ┆ …         ┆ …         ┆ …         ┆ …         │
│ FAM86B2 ┆ 5.615606   ┆ 5.311966  ┆ 5.0533    ┆ … ┆ 5.146797  ┆ 5.129829  ┆ 5.213772  ┆ 5.857212  │
│ FAM86FP ┆ 5.615606   ┆ 5.311966  ┆ 5.0533    ┆ … ┆ 5.146797  ┆ 5.129829  ┆ 5.213772  ┆ 5.857212  │
│ FAM86B1 ┆ 5.615606   ┆ 5.311966  ┆ 5.0533    ┆ … ┆ 5.146797  ┆ 5.129829  ┆ 5.213772  ┆ 5.857212  │
│ FAM86C1 ┆ 5.615606   ┆ 5.311966  ┆ 5.0533    ┆ … ┆ 5.146797  ┆ 5.129829  ┆ 5.213772  ┆ 5.857212  │
└─────────┴────────────┴───────────┴───────────┴───┴───────────┴───────────┴───────────┴───────────┘

In [10]:
GSE95065_small = pl.read_parquet("../data/adex-database/samples/archived_not_used/GSE95065_2.tsv.parquet")
GSE95065_small

shape: (39, 34)
┌───────────┬───────────┬───────────┬───────────┬───┬───────────┬───────────┬───────────┬──────────┐
│ gene      ┆ GSM249559 ┆ GSM249559 ┆ GSM249559 ┆ … ┆ GSM249562 ┆ GSM249562 ┆ GSM249562 ┆ GSM24956 │
│ ---       ┆ 4         ┆ 5         ┆ 6         ┆   ┆ 3         ┆ 4         ┆ 5         ┆ 26       │
│ str       ┆ ---       ┆ ---       ┆ ---       ┆   ┆ ---       ┆ ---       ┆ ---       ┆ ---      │
│           ┆ f64       ┆ f64       ┆ f64       ┆   ┆ f64       ┆ f64       ┆ f64       ┆ f64      │
╞═══════════╪═══════════╪═══════════╪═══════════╪═══╪═══════════╪═══════════╪═══════════╪══════════╡
│ CEBPE     ┆ 7.430844  ┆ 7.424512  ┆ 7.380993  ┆ … ┆ 7.365737  ┆ 7.076975  ┆ 7.140006  ┆ 7.11732  │
│ ADCYAP1R1 ┆ 5.94925   ┆ 5.912904  ┆ 5.952583  ┆ … ┆ 5.5735    ┆ 6.109171  ┆ 5.765169  ┆ 5.834804 │
│ COL7A1    ┆ 6.533028  ┆ 6.441243  ┆ 6.26865   ┆ … ┆ 6.277183  ┆ 6.302032  ┆ 6.382896  ┆ 6.893414 │
│ KLF6      ┆ 6.90694   ┆ 6.48762   ┆ 6.229516  ┆ … ┆ 5.903811  ┆ 6.093964  ┆ 5.95448   ┆ 6.172006 │
│ …         ┆ …         ┆ …         ┆ …         ┆ … ┆ …         ┆ …         ┆ …         ┆ …        │
│ SEMA4G    ┆ 6.940605  ┆ 6.929228  ┆ 6.982693  ┆ … ┆ 6.655612  ┆ 6.445045  ┆ 6.463942  ┆ 6.871585 │
│ ELAC2     ┆ 6.794238  ┆ 6.901478  ┆ 6.877149  ┆ … ┆ 7.007394  ┆ 7.212608  ┆ 6.876413  ┆ 7.353629 │
│ TMEM168   ┆ 8.103874  ┆ 7.777616  ┆ 7.954945  ┆ … ┆ 8.053123  ┆ 8.217889  ┆ 8.097572  ┆ 8.335969 │
│ CAPN1     ┆ 7.074772  ┆ 7.222501  ┆ 7.121602  ┆ … ┆ 7.052602  ┆ 6.754836  ┆ 7.132416  ┆ 6.856021 │
└───────────┴───────────┴───────────┴───────────┴───┴───────────┴───────────┴───────────┴──────────┘

In [20]:
# Make certain that we have the same samples in the two files 
large_set_samples = set(GSE95065_large.columns)
small_set_samples = set(GSE95065_small.columns)
large_set_samples == small_set_samples

True

In [23]:
# Make certain that there are no duplicated genes between the two files 
large_set_genes = set(GSE95065_large.select("gene").to_series().to_list())
small_set_genes = set(GSE95065_small.select("gene").to_series().to_list())
large_set_genes.intersection(small_set_genes)

{'ASAP3',
 'CAPN1',
 'CKLF',
 'COL7A1',
 'CPA4',
 'CSF2RA',
 'CTBP1',
 'DIAPH1',
 'ELAC2',
 'FAM171A1',
 'GLTP',
 'HIF1AN',
 'IFT57',
 'IPO9',
 'KLF6',
 'LEF1',
 'LUC7L',
 'PBK',
 'PXDC1',
 'SEMA4G',
 'SLC52A1',
 'TMEM168',
 'TOR1A',
 'TTC12',
 'TUBB',
 'UNC5B',
 'WDYHV1',
 'ZC3HAV1',
 'ZCCHC24'}

There seem to be common genes between the two files which does not make much sense since it would be different readings for the same sample and gene. 
Will discard the small file instead of the old decision that was keeping it.
Commenting the lines below that were creating the merged dataset.  

In [13]:
# GSE95065_merged = pl.concat([GSE95065_small, GSE95065_large], rechunk=True)
# GSE95065_merged

shape: (10_516, 34)
┌───────────┬───────────┬───────────┬───────────┬───┬───────────┬───────────┬───────────┬──────────┐
│ gene      ┆ GSM249559 ┆ GSM249559 ┆ GSM249559 ┆ … ┆ GSM249562 ┆ GSM249562 ┆ GSM249562 ┆ GSM24956 │
│ ---       ┆ 4         ┆ 5         ┆ 6         ┆   ┆ 3         ┆ 4         ┆ 5         ┆ 26       │
│ str       ┆ ---       ┆ ---       ┆ ---       ┆   ┆ ---       ┆ ---       ┆ ---       ┆ ---      │
│           ┆ f64       ┆ f64       ┆ f64       ┆   ┆ f64       ┆ f64       ┆ f64       ┆ f64      │
╞═══════════╪═══════════╪═══════════╪═══════════╪═══╪═══════════╪═══════════╪═══════════╪══════════╡
│ CEBPE     ┆ 7.430844  ┆ 7.424512  ┆ 7.380993  ┆ … ┆ 7.365737  ┆ 7.076975  ┆ 7.140006  ┆ 7.11732  │
│ ADCYAP1R1 ┆ 5.94925   ┆ 5.912904  ┆ 5.952583  ┆ … ┆ 5.5735    ┆ 6.109171  ┆ 5.765169  ┆ 5.834804 │
│ COL7A1    ┆ 6.533028  ┆ 6.441243  ┆ 6.26865   ┆ … ┆ 6.277183  ┆ 6.302032  ┆ 6.382896  ┆ 6.893414 │
│ KLF6      ┆ 6.90694   ┆ 6.48762   ┆ 6.229516  ┆ … ┆ 5.903811  ┆ 6.093964  ┆ 5.95448   ┆ 6.172006 │
│ …         ┆ …         ┆ …         ┆ …         ┆ … ┆ …         ┆ …         ┆ …         ┆ …        │
│ FAM86B2   ┆ 5.615606  ┆ 5.311966  ┆ 5.0533    ┆ … ┆ 5.146797  ┆ 5.129829  ┆ 5.213772  ┆ 5.857212 │
│ FAM86FP   ┆ 5.615606  ┆ 5.311966  ┆ 5.0533    ┆ … ┆ 5.146797  ┆ 5.129829  ┆ 5.213772  ┆ 5.857212 │
│ FAM86B1   ┆ 5.615606  ┆ 5.311966  ┆ 5.0533    ┆ … ┆ 5.146797  ┆ 5.129829  ┆ 5.213772  ┆ 5.857212 │
│ FAM86C1   ┆ 5.615606  ┆ 5.311966  ┆ 5.0533    ┆ … ┆ 5.146797  ┆ 5.129829  ┆ 5.213772  ┆ 5.857212 │
└───────────┴───────────┴───────────┴───────────┴───┴───────────┴───────────┴───────────┴──────────┘

In [15]:
# GSE95065_merged.write_parquet(file="../data/adex-database/samples/SSc/GSE95065_merged.tsv.parquet", use_pyarrow=True)

In [4]:
# validate 
# pl.read_parquet("../data/adex-database/samples/SSc/GSE95065_merged.tsv.parquet")

shape: (10_516, 34)
┌───────────┬───────────┬───────────┬───────────┬───┬───────────┬───────────┬───────────┬──────────┐
│ gene      ┆ GSM249559 ┆ GSM249559 ┆ GSM249559 ┆ … ┆ GSM249562 ┆ GSM249562 ┆ GSM249562 ┆ GSM24956 │
│ ---       ┆ 4         ┆ 5         ┆ 6         ┆   ┆ 3         ┆ 4         ┆ 5         ┆ 26       │
│ str       ┆ ---       ┆ ---       ┆ ---       ┆   ┆ ---       ┆ ---       ┆ ---       ┆ ---      │
│           ┆ f64       ┆ f64       ┆ f64       ┆   ┆ f64       ┆ f64       ┆ f64       ┆ f64      │
╞═══════════╪═══════════╪═══════════╪═══════════╪═══╪═══════════╪═══════════╪═══════════╪══════════╡
│ CEBPE     ┆ 7.430844  ┆ 7.424512  ┆ 7.380993  ┆ … ┆ 7.365737  ┆ 7.076975  ┆ 7.140006  ┆ 7.11732  │
│ ADCYAP1R1 ┆ 5.94925   ┆ 5.912904  ┆ 5.952583  ┆ … ┆ 5.5735    ┆ 6.109171  ┆ 5.765169  ┆ 5.834804 │
│ COL7A1    ┆ 6.533028  ┆ 6.441243  ┆ 6.26865   ┆ … ┆ 6.277183  ┆ 6.302032  ┆ 6.382896  ┆ 6.893414 │
│ KLF6      ┆ 6.90694   ┆ 6.48762   ┆ 6.229516  ┆ … ┆ 5.903811  ┆ 6.093964  ┆ 5.95448   ┆ 6.172006 │
│ …         ┆ …         ┆ …         ┆ …         ┆ … ┆ …         ┆ …         ┆ …         ┆ …        │
│ FAM86B2   ┆ 5.615606  ┆ 5.311966  ┆ 5.0533    ┆ … ┆ 5.146797  ┆ 5.129829  ┆ 5.213772  ┆ 5.857212 │
│ FAM86FP   ┆ 5.615606  ┆ 5.311966  ┆ 5.0533    ┆ … ┆ 5.146797  ┆ 5.129829  ┆ 5.213772  ┆ 5.857212 │
│ FAM86B1   ┆ 5.615606  ┆ 5.311966  ┆ 5.0533    ┆ … ┆ 5.146797  ┆ 5.129829  ┆ 5.213772  ┆ 5.857212 │
│ FAM86C1   ┆ 5.615606  ┆ 5.311966  ┆ 5.0533    ┆ … ┆ 5.146797  ┆ 5.129829  ┆ 5.213772  ┆ 5.857212 │
└───────────┴───────────┴───────────┴───────────┴───┴───────────┴───────────┴───────────┴──────────┘

### GSE55098

The second file seems to only have 5 samples. Easier to just discard it. This whole dataset is too small anyway. 

In [3]:
GSE55098_one = pl.read_parquet("../data/adex-database/samples/GSE55098.tsv.parquet")
GSE55098_two = pl.read_parquet("../data/adex-database/samples/GSE55098_2.tsv.parquet")

In [4]:
GSE55098_one

shape: (22_803, 23)
┌─────────┬────────────┬───────────┬───────────┬───┬───────────┬───────────┬───────────┬───────────┐
│ gene    ┆ GSM1329616 ┆ GSM132961 ┆ GSM132961 ┆ … ┆ GSM132963 ┆ GSM132963 ┆ GSM132963 ┆ GSM132963 │
│ ---     ┆ ---        ┆ 7         ┆ 8         ┆   ┆ 4         ┆ 5         ┆ 6         ┆ 7         │
│ str     ┆ f64        ┆ ---       ┆ ---       ┆   ┆ ---       ┆ ---       ┆ ---       ┆ ---       │
│         ┆            ┆ f64       ┆ f64       ┆   ┆ f64       ┆ f64       ┆ f64       ┆ f64       │
╞═════════╪════════════╪═══════════╪═══════════╪═══╪═══════════╪═══════════╪═══════════╪═══════════╡
│ RFC2    ┆ 7.29312    ┆ 7.478815  ┆ 7.163901  ┆ … ┆ 7.331175  ┆ 7.392728  ┆ 7.518944  ┆ 7.957457  │
│ HSPA6   ┆ 8.646147   ┆ 9.134429  ┆ 7.398783  ┆ … ┆ 8.566601  ┆ 10.476805 ┆ 8.968847  ┆ 9.544259  │
│ PAX8    ┆ 4.955674   ┆ 4.633809  ┆ 4.779538  ┆ … ┆ 4.845228  ┆ 4.768793  ┆ 4.982103  ┆ 5.013744  │
│ GUCA1A  ┆ 4.566657   ┆ 4.454742  ┆ 4.474081  ┆ … ┆ 4.466288  ┆ 4.467433  ┆ 4.790837  ┆ 4.408654  │
│ …       ┆ …          ┆ …         ┆ …         ┆ … ┆ …         ┆ …         ┆ …         ┆ …         │
│ FAM86DP ┆ 4.879327   ┆ 4.798487  ┆ 5.032185  ┆ … ┆ 4.911538  ┆ 4.787081  ┆ 5.107459  ┆ 5.008376  │
│ FAM86B2 ┆ 4.879327   ┆ 4.798487  ┆ 5.032185  ┆ … ┆ 4.911538  ┆ 4.787081  ┆ 5.107459  ┆ 5.008376  │
│ FAM86FP ┆ 4.879327   ┆ 4.798487  ┆ 5.032185  ┆ … ┆ 4.911538  ┆ 4.787081  ┆ 5.107459  ┆ 5.008376  │
│ FAM86B1 ┆ 4.879327   ┆ 4.798487  ┆ 5.032185  ┆ … ┆ 4.911538  ┆ 4.787081  ┆ 5.107459  ┆ 5.008376  │
└─────────┴────────────┴───────────┴───────────┴───┴───────────┴───────────┴───────────┴───────────┘

In [5]:
GSE55098_two

shape: (22_803, 6)
┌─────────┬────────────┬────────────┬────────────┬────────────┬────────────┐
│ gene    ┆ GSM1329617 ┆ GSM1329622 ┆ GSM1329627 ┆ GSM1329632 ┆ GSM1329637 │
│ ---     ┆ ---        ┆ ---        ┆ ---        ┆ ---        ┆ ---        │
│ str     ┆ f64        ┆ f64        ┆ f64        ┆ f64        ┆ f64        │
╞═════════╪════════════╪════════════╪════════════╪════════════╪════════════╡
│ RFC2    ┆ 7.737012   ┆ 7.836727   ┆ 7.794087   ┆ 7.432892   ┆ 8.124732   │
│ HSPA6   ┆ 9.286261   ┆ 8.037874   ┆ 8.922395   ┆ 8.983894   ┆ 9.658149   │
│ PAX8    ┆ 4.728237   ┆ 5.001011   ┆ 4.880428   ┆ 4.856147   ┆ 4.956531   │
│ GUCA1A  ┆ 4.445629   ┆ 4.519743   ┆ 4.519743   ┆ 4.524788   ┆ 4.519743   │
│ …       ┆ …          ┆ …          ┆ …          ┆ …          ┆ …          │
│ FAM86DP ┆ 5.276182   ┆ 5.248915   ┆ 5.081886   ┆ 5.771317   ┆ 5.43288    │
│ FAM86B2 ┆ 5.276182   ┆ 5.248915   ┆ 5.081886   ┆ 5.771317   ┆ 5.43288    │
│ FAM86FP ┆ 5.276182   ┆ 5.248915   ┆ 5.081886   ┆ 5.771317   ┆ 5.43288    │
│ FAM86B1 ┆ 5.276182   ┆ 5.248915   ┆ 5.081886   ┆ 5.771317   ┆ 5.43288    │
└─────────┴────────────┴────────────┴────────────┴────────────┴────────────┘

# Determine common genes per condition

The datasets for each condition come from different sources. 
To use all the samples of a condition together, we have to see which are the common genes.

In [10]:
ra_common_genes = gene_intersection(ra_data)
t1d_common_genes = gene_intersection(t1d_data)
ssc_common_genes = gene_intersection(ssc_data)
sle_common_genes = gene_intersection(sle_data)
sjs_common_genes = gene_intersection(sjs_data)

print(f"""
    Common Genes:  
    
    RA: {len(ra_common_genes)}
    T1D: {len(t1d_common_genes)}
    SSc: {len(ssc_common_genes)}
    SLE: {len(sle_common_genes)}
    SjS: {len(sjs_common_genes)}
""")


    Common Genes:  
    
    RA: 2602
    T1D: 2504
    SSc: 6582
    SLE: 2050
    SjS: 7683


In [11]:
ra_cleaned_df = common_genes_dataframe(ra_data)
t1d_cleaned_df = common_genes_dataframe(t1d_data)
ssc_cleaned_df = common_genes_dataframe(ssc_data)
sle_cleaned_df = common_genes_dataframe(sle_data)
sjs_cleaned_df = common_genes_dataframe(sjs_data)

print(f"""
    Shape (including only common genes): 
    
    RA: {ra_cleaned_df.shape}
    T1D: {t1d_cleaned_df.shape}
    SSc: {ssc_cleaned_df.shape}
    SLE: {sle_cleaned_df.shape}
    SjS: {sjs_cleaned_df.shape}
""")



    Shape (including only common genes): 
    
    RA: (2602, 1123)
    T1D: (2504, 177)
    SSc: (6582, 230)
    SLE: (2050, 2054)
    SjS: (7683, 401)


In [12]:
ra_cleaned_df

shape: (2_602, 1_123)
┌───────────┬───────────┬───────────┬───────────┬───┬───────────┬───────────┬───────────┬──────────┐
│ gene      ┆ GSM239736 ┆ GSM239736 ┆ GSM239737 ┆ … ┆ GSM133732 ┆ GSM133732 ┆ GSM133732 ┆ GSM13373 │
│ ---       ┆ 8         ┆ 9         ┆ 0         ┆   ┆ 3         ┆ 4         ┆ 5         ┆ 26       │
│ str       ┆ ---       ┆ ---       ┆ ---       ┆   ┆ ---       ┆ ---       ┆ ---       ┆ ---      │
│           ┆ f64       ┆ f64       ┆ f64       ┆   ┆ f64       ┆ f64       ┆ f64       ┆ f64      │
╞═══════════╪═══════════╪═══════════╪═══════════╪═══╪═══════════╪═══════════╪═══════════╪══════════╡
│ PRPF8     ┆ 10.300174 ┆ 10.035596 ┆ 9.946043  ┆ … ┆ 10.431685 ┆ 10.290099 ┆ 10.265424 ┆ 10.20004 │
│ CAPNS1    ┆ 10.798511 ┆ 10.615747 ┆ 10.336886 ┆ … ┆ 12.097445 ┆ 12.521696 ┆ 12.38552  ┆ 12.34751 │
│           ┆           ┆           ┆           ┆   ┆           ┆           ┆           ┆ 7        │
│ RPL35     ┆ 9.626852  ┆ 9.906102  ┆ 9.301532  ┆ … ┆ 12.842306 ┆ 12.761665 ┆ 12.597323 ┆ 12.03921 │
│           ┆           ┆           ┆           ┆   ┆           ┆           ┆           ┆ 1        │
│ EIF4G2    ┆ 10.906887 ┆ 11.034909 ┆ 11.479914 ┆ … ┆ 8.729406  ┆ 8.821131  ┆ 8.823763  ┆ 8.914625 │
│ EIF3D     ┆ 10.012536 ┆ 10.060014 ┆ 9.672631  ┆ … ┆ 11.175461 ┆ 11.318008 ┆ 11.130285 ┆ 11.04941 │
│           ┆           ┆           ┆           ┆   ┆           ┆           ┆           ┆ 5        │
│ …         ┆ …         ┆ …         ┆ …         ┆ … ┆ …         ┆ …         ┆ …         ┆ …        │
│ AGAP6     ┆ 6.531409  ┆ 6.875907  ┆ 6.816773  ┆ … ┆ 9.720059  ┆ 10.091315 ┆ 9.854061  ┆ 9.590049 │
│ FBXO41    ┆ 7.010478  ┆ 7.157273  ┆ 7.0324    ┆ … ┆ 7.361597  ┆ 7.150914  ┆ 7.276689  ┆ 7.565414 │
│ LINC00869 ┆ 7.262123  ┆ 7.219805  ┆ 7.326638  ┆ … ┆ 9.875226  ┆ 9.298062  ┆ 9.574527  ┆ 9.206733 │
│ PFDN6     ┆ 6.774082  ┆ 6.761453  ┆ 6.662362  ┆ … ┆ 7.469991  ┆ 7.314559  ┆ 7.233719  ┆ 7.243504 │
│ SDHAP1    ┆ 5.934374  ┆ 5.932378  ┆ 6.341597  ┆ … ┆ 10.71353  ┆ 10.613987 ┆ 10.680925 ┆ 10.50665 │
│           ┆           ┆           ┆           ┆   ┆           ┆           ┆           ┆ 6        │
└───────────┴───────────┴───────────┴───────────┴───┴───────────┴───────────┴───────────┴──────────┘

In [13]:
total_samples = ra_cleaned_df.shape[1] + t1d_cleaned_df.shape[1] + ssc_cleaned_df.shape[1] + sle_cleaned_df.shape[1] + sjs_cleaned_df.shape[1]
total_samples

3985

Samples seem to be a bit more than the unique ones that appear in the metadata file which is `3854`.
Possibly we need to filter out the extras because for these we have no metadata.  

# Determine frequent genes per group of datasets

The datasets for each condition come from different sources. 
Explore using samples of a condition together, but keeping only the genes that appear in 80% of the samples. 
This is the same logic as the above, but relaxes the constraint and keeping the high-frequency genes instead of just the common ones. 
So it allows for a certain percent of null values.  

In [14]:
# Experiment 
filtered = (
    high_frequency_genes_dataframe(
        dataframes=load_data_per_condition(Condition.SLE, "../data/adex-database/samples"),
        allowed_null_percentage=1,  # will filter later, no filtering here
        drop_frequencies_column=False
    )
    .with_columns(
        pl.sum_horizontal(pl.all().is_null() / pl.all().count())
        .alias("Null-Percentage")
    )
    .filter(pl.col("Null-Percentage") <= 0.01)
)

filtered

shape: (5_300, 2_055)
┌─────────┬────────────┬───────────┬───────────┬───┬───────────┬───────────┬───────────┬───────────┐
│ gene    ┆ GSM2159829 ┆ GSM215983 ┆ GSM215983 ┆ … ┆ GSM186374 ┆ GSM186374 ┆ GSM186374 ┆ Null-Perc │
│ ---     ┆ _GSM218666 ┆ 0_GSM2186 ┆ 1_GSM2186 ┆   ┆ 7         ┆ 8         ┆ 9         ┆ entage    │
│ str     ┆ 8          ┆ 669       ┆ 670       ┆   ┆ ---       ┆ ---       ┆ ---       ┆ ---       │
│         ┆ ---        ┆ ---       ┆ ---       ┆   ┆ f64       ┆ f64       ┆ f64       ┆ f64       │
│         ┆ f64        ┆ f64       ┆ f64       ┆   ┆           ┆           ┆           ┆           │
╞═════════╪════════════╪═══════════╪═══════════╪═══╪═══════════╪═══════════╪═══════════╪═══════════╡
│ EEF1A1  ┆ 14.727188  ┆ 14.795797 ┆ 14.817517 ┆ … ┆ 17.825161 ┆ 17.79348  ┆ 18.881821 ┆ 0.0       │
│ GAPDH   ┆ 11.708472  ┆ 11.65505  ┆ 12.340323 ┆ … ┆ 16.026341 ┆ 16.415398 ┆ 15.953825 ┆ 0.0       │
│ RPS28   ┆ 12.334125  ┆ 12.684471 ┆ 12.404128 ┆ … ┆ 13.947265 ┆ 13.731703 ┆ 14.274837 ┆ 0.005341  │
│ LPP     ┆ 10.270518  ┆ 10.157473 ┆ 9.924002  ┆ … ┆ 11.650111 ┆ 11.814043 ┆ 11.186168 ┆ 0.0       │
│ HNRNPAB ┆ 9.428744   ┆ 9.626779  ┆ 9.24285   ┆ … ┆ 11.662398 ┆ 11.68726  ┆ 11.419868 ┆ 0.005341  │
│ …       ┆ …          ┆ …         ┆ …         ┆ … ┆ …         ┆ …         ┆ …         ┆ …         │
│ TAOK3   ┆ 8.228441   ┆ 8.161298  ┆ 8.146965  ┆ … ┆ 12.675293 ┆ 12.634832 ┆ 12.097556 ┆ 0.009191  │
│ GRK2    ┆ 12.278163  ┆ 12.391146 ┆ 12.598222 ┆ … ┆ 15.689227 ┆ 15.423748 ┆ 15.720508 ┆ 0.009191  │
│ DHRS9   ┆ null       ┆ null      ┆ null      ┆ … ┆ 10.209494 ┆ 12.137662 ┆ 10.969615 ┆ 0.004745  │
│ IGF1R   ┆ null       ┆ null      ┆ null      ┆ … ┆ 12.841821 ┆ 12.84864  ┆ 12.388443 ┆ 0.003814  │
│ TECPR2  ┆ null       ┆ null      ┆ null      ┆ … ┆ 13.479952 ┆ 13.430435 ┆ 12.737796 ┆ 0.009155  │
└─────────┴────────────┴───────────┴───────────┴───┴───────────┴───────────┴───────────┴───────────┘

In [15]:
print(f"""
    Shape (all data): 
    RA: {high_frequency_genes_dataframe(ra_data, allowed_null_percentage=1, drop_frequencies_column=True).shape}
    T1D: {high_frequency_genes_dataframe(t1d_data, allowed_null_percentage=1, drop_frequencies_column=True).shape}
    SSc: {high_frequency_genes_dataframe(ssc_data, allowed_null_percentage=1, drop_frequencies_column=True).shape}
    SLE: {high_frequency_genes_dataframe(sle_data, allowed_null_percentage=1, drop_frequencies_column=True).shape}
    SjS: {high_frequency_genes_dataframe(sjs_data, allowed_null_percentage=1, drop_frequencies_column=True).shape}
""")

print(f"""
    Shape (only commom genes): 
    RA: {high_frequency_genes_dataframe(ra_data, allowed_null_percentage=0, drop_frequencies_column=True).shape}
    T1D: {high_frequency_genes_dataframe(t1d_data, allowed_null_percentage=0, drop_frequencies_column=True).shape}
    SSc: {high_frequency_genes_dataframe(ssc_data, allowed_null_percentage=0, drop_frequencies_column=True).shape}
    SLE: {high_frequency_genes_dataframe(sle_data, allowed_null_percentage=0, drop_frequencies_column=True).shape}
    SjS: {high_frequency_genes_dataframe(sjs_data, allowed_null_percentage=0, drop_frequencies_column=True).shape}
""")

allowed_null_percentage = 0.2

print(f"""
    Shape (including only high-frequency genes (genes that appear in 80% of the samples)): 
    RA: {high_frequency_genes_dataframe(ra_data, allowed_null_percentage=allowed_null_percentage, drop_frequencies_column=True).shape}
    T1D: {high_frequency_genes_dataframe(t1d_data, allowed_null_percentage=allowed_null_percentage, drop_frequencies_column=True).shape}
    SSc: {high_frequency_genes_dataframe(ssc_data, allowed_null_percentage=allowed_null_percentage, drop_frequencies_column=True).shape}
    SLE: {high_frequency_genes_dataframe(sle_data, allowed_null_percentage=allowed_null_percentage, drop_frequencies_column=True).shape}
    SjS: {high_frequency_genes_dataframe(sjs_data, allowed_null_percentage=allowed_null_percentage, drop_frequencies_column=True).shape}
""")


    Shape (all data): 
    RA: (28661, 1123)
    T1D: (27581, 177)
    SSc: (20411, 230)
    SLE: (27803, 2054)
    SjS: (23144, 401)

    Shape (only commom genes): 
    RA: (2602, 1123)
    T1D: (2504, 177)
    SSc: (6582, 230)
    SLE: (2050, 2054)
    SjS: (7683, 401)

    Shape (including only high-frequency genes (genes that appear in 80% of the samples)): 
    RA: (28661, 1123)
    T1D: (27581, 177)
    SSc: (20411, 230)
    SLE: (27803, 2054)
    SjS: (23144, 401)


> Observation: Allowing 20% of null values is keeping the whole dataset  

# Available tissues per condition

In [3]:
for condition in Condition:
    dataset = get_pre_processed_dataset(ConditionDataLoader(condition), files_path, metadata_path, datasets_info_path)
    print(f"{condition.name}: {dataset.select('Tissue').unique()}")

RA: shape: (3, 1)
┌───────────────────┐
│ Tissue            │
│ ---               │
│ str               │
╞═══════════════════╡
│ Peripheral blood  │
│ Whole blood       │
│ Synovial membrane │
└───────────────────┘
T1D: shape: (2, 1)
┌──────────────────┐
│ Tissue           │
│ ---              │
│ str              │
╞══════════════════╡
│ Whole blood      │
│ Peripheral blood │
└──────────────────┘
SSc: shape: (3, 1)
┌──────────────────┐
│ Tissue           │
│ ---              │
│ str              │
╞══════════════════╡
│ Whole blood      │
│ Peripheral blood │
│ Skin             │
└──────────────────┘
SLE: shape: (3, 1)
┌──────────────────┐
│ Tissue           │
│ ---              │
│ str              │
╞══════════════════╡
│ Peripheral blood │
│ Skin             │
│ Whole blood      │
└──────────────────┘
SjS: shape: (5, 1)
┌──────────────────┐
│ Tissue           │
│ ---              │
│ str              │
╞══════════════════╡
│ Salivary gland   │
│ Saliva           │
│ Whole blood  

# Sequencing Technique (Microarrays vs RNA-Seq) 

In [17]:
datasets_info = pl.read_csv(datasets_info_path)
datasets_info.shape

(41, 5)

In [18]:
datasets_info

shape: (41, 5)
┌───────────┬─────────┬─────────────────────────────────┬────────────────────────────────┬─────────┐
│ Dataset   ┆ Disease ┆ Method                          ┆ Title                          ┆ Samples │
│ ---       ┆ ---     ┆ ---                             ┆ ---                            ┆ ---     │
│ str       ┆ str     ┆ str                             ┆ str                            ┆ i64     │
╞═══════════╪═════════╪═════════════════════════════════╪════════════════════════════════╪═════════╡
│ GSE10325  ┆ SLE     ┆ Expression profiling by array   ┆ Combined deficiency of         ┆ 67      │
│           ┆         ┆                                 ┆ proapopto…                     ┆         │
│ GSE104174 ┆ SSc     ┆ Expression profiling by high    ┆ Changes in macrophage          ┆ 72      │
│           ┆         ┆ thr…                            ┆ transcript…                    ┆         │
│ GSE108497 ┆ SLE     ┆ Expression profiling by array   ┆ Illumina HumanHT-12 V4.0       ┆ 512     │
│           ┆         ┆                                 ┆ express…                       ┆         │
│ GSE110169 ┆ SLE RA  ┆ Expression profiling by array   ┆ [HG-U219] Affymetrix Human     ┆ 234     │
│           ┆         ┆                                 ┆ Genom…                         ┆         │
│ GSE110174 ┆ SLE     ┆ Expression profiling by array   ┆ [HT_HG-U133_Plus_PM]           ┆ 154     │
│           ┆         ┆                                 ┆ Affymetrix …                   ┆         │
│ …         ┆ …       ┆ …                               ┆ …                              ┆ …       │
│ GSE84844  ┆ SjS     ┆ Expression profiling by array   ┆ Multiomic disease signatures   ┆ 60      │
│           ┆         ┆                                 ┆ con…                           ┆         │
│ GSE89408  ┆ RA      ┆ Expression profiling by high    ┆ CD40L-Dependent Pathway Is     ┆ 218     │
│           ┆         ┆ thr…                            ┆ Activ…                         ┆         │
│ GSE90081  ┆ RA      ┆ Expression profiling by high    ┆ Discovery of new candidate     ┆ 24      │
│           ┆         ┆ thr…                            ┆ genes…                         ┆         │
│ GSE93683  ┆ SjS     ┆ Expression profiling by array   ┆ Multiomic disease signatures   ┆ 48      │
│           ┆         ┆                                 ┆ con…                           ┆         │
│ GSE95065  ┆ SSc     ┆ Expression profiling by array   ┆ [HG-U133A_2] Affymetrix Human  ┆ 33      │
│           ┆         ┆                                 ┆ Ge…                            ┆         │
└───────────┴─────────┴─────────────────────────────────┴────────────────────────────────┴─────────┘

In [20]:
datasets_info.group_by("Method").len()

shape: (2, 2)
┌───────────────────────────────────┬─────┐
│ Method                            ┆ len │
│ ---                               ┆ --- │
│ str                               ┆ u32 │
╞═══════════════════════════════════╪═════╡
│ Expression profiling by array     ┆ 31  │
│ Expression profiling by high thr… ┆ 10  │
└───────────────────────────────────┴─────┘

# Per Condition/Sequencing/Tissue stats 

## Testing: 

In [4]:
# Test
sample_df = (
    get_pre_processed_dataset(
        ConditionSequencingTissueDataLoader(
            Condition.SjS, 
            SequencingTechnique.MICROARRAYS, 
            TissueEnum.SALIVA
        ), 
        files_path, 
        metadata_path, 
        datasets_info_path, 
        return_metadata=False
    )
)

sample_df.shape

(20, 20400)

In [6]:
sample_df.head(20)

shape: (20, 20_400)
┌───────────┬───────────┬───────────┬───────────┬───┬───────────┬───────────┬───────────┬──────────┐
│ Sample    ┆ RFC2      ┆ HSPA6     ┆ PAX8      ┆ … ┆ FAM86DP   ┆ FAM86B2   ┆ FAM86FP   ┆ FAM86B1  │
│ ---       ┆ ---       ┆ ---       ┆ ---       ┆   ┆ ---       ┆ ---       ┆ ---       ┆ ---      │
│ str       ┆ str       ┆ str       ┆ str       ┆   ┆ str       ┆ str       ┆ str       ┆ str      │
╞═══════════╪═══════════╪═══════════╪═══════════╪═══╪═══════════╪═══════════╪═══════════╪══════════╡
│ GSM180372 ┆ 1.9490484 ┆ 2.0559096 ┆ 1.8692430 ┆ … ┆ 1.7254567 ┆ 1.7254567 ┆ 1.7254567 ┆ 1.725456 │
│           ┆ 3778271   ┆ 2074666   ┆ 3772085   ┆   ┆ 3161155   ┆ 3161155   ┆ 3161155   ┆ 73161155 │
│ GSM180380 ┆ 2.2064759 ┆ 2.3095848 ┆ 1.8555991 ┆ … ┆ 1.9008513 ┆ 1.9008513 ┆ 1.9008513 ┆ 1.900851 │
│           ┆ 3790151   ┆ 4156738   ┆ 6246481   ┆   ┆ 5443914   ┆ 5443914   ┆ 5443914   ┆ 35443914 │
│ GSM180383 ┆ 1.9698279 ┆ 3.5230307 ┆ 1.9122797 ┆ … ┆ 1.8314809 ┆ 1.8314809 ┆ 1.8314809 ┆ 1.831480 │
│           ┆ 9784334   ┆ 2479728   ┆ 5022177   ┆   ┆ 432738    ┆ 432738    ┆ 432738    ┆ 9432738  │
│ GSM180378 ┆ 1.8669118 ┆ 2.2246756 ┆ 1.9331012 ┆ … ┆ 1.8620079 ┆ 1.8620079 ┆ 1.8620079 ┆ 1.862007 │
│           ┆ 5803368   ┆ 3603498   ┆ 6297283   ┆   ┆ 734293    ┆ 734293    ┆ 734293    ┆ 9734293  │
│ GSM180379 ┆ 2.0565130 ┆ 3.6120671 ┆ 1.9251530 ┆ … ┆ 1.7696210 ┆ 1.7696210 ┆ 1.7696210 ┆ 1.769621 │
│           ┆ 7727808   ┆ 6693603   ┆ 0289088   ┆   ┆ 3814194   ┆ 3814194   ┆ 3814194   ┆ 03814194 │
│ …         ┆ …         ┆ …         ┆ …         ┆ … ┆ …         ┆ …         ┆ …         ┆ …        │
│ GSM180382 ┆ 1.9159526 ┆ 2.8009246 ┆ 1.9084982 ┆ … ┆ 1.7674132 ┆ 1.7674132 ┆ 1.7674132 ┆ 1.767413 │
│           ┆ 8165859   ┆ 7664153   ┆ 2087322   ┆   ┆ 0670185   ┆ 0670185   ┆ 0670185   ┆ 20670185 │
│ GSM180386 ┆ 1.8285657 ┆ 2.8924206 ┆ 1.9306246 ┆ … ┆ 1.7933174 ┆ 1.7933174 ┆ 1.7933174 ┆ 1.793317 │
│           ┆ 0472821   ┆ 7968546   ┆ 7969617   ┆   ┆ 7834362   ┆ 7834362   ┆ 7834362   ┆ 47834362 │
│ GSM180387 ┆ 1.8832300 ┆ 2.3298980 ┆ 1.9013964 ┆ … ┆ 1.8320456 ┆ 1.8320456 ┆ 1.8320456 ┆ 1.832045 │
│           ┆ 9021046   ┆ 8686946   ┆ 2802121   ┆   ┆ 4330272   ┆ 4330272   ┆ 4330272   ┆ 64330272 │
│ GSM180373 ┆ 1.9213025 ┆ 2.1731179 ┆ 1.8768787 ┆ … ┆ 1.8601290 ┆ 1.8601290 ┆ 1.8601290 ┆ 1.860129 │
│           ┆ 3152473   ┆ 6290032   ┆ 5814847   ┆   ┆ 5036111   ┆ 5036111   ┆ 5036111   ┆ 05036111 │
│ GSM180388 ┆ 1.9017108 ┆ 2.3333943 ┆ 1.9502601 ┆ … ┆ 1.9624214 ┆ 1.9624214 ┆ 1.9624214 ┆ 1.962421 │
│           ┆ 5356204   ┆ 6234204   ┆ 263104    ┆   ┆ 2404516   ┆ 2404516   ┆ 2404516   ┆ 42404516 │
└───────────┴───────────┴───────────┴───────────┴───┴───────────┴───────────┴───────────┴──────────┘

## Stats

In [7]:
for sequencing_technique in SequencingTechnique: 
    for condition in Condition:
        for tissue in TissueEnum: 
            dataset = get_pre_processed_dataset(
                ConditionSequencingTissueDataLoader(
                    condition, 
                    sequencing_technique, 
                    tissue
                ), 
                files_path, 
                metadata_path, 
                datasets_info_path,
                return_metadata=False
            )
            
            if dataset is not None:
                shape = dataset.shape
                print(f"{sequencing_technique.name}|{condition.name}|{tissue.name}: Samples: {shape[0]}, Genes: {shape[1] - 1}")  # -1 in genes because first column is the Sample 

MICROARRAYS|RA|PERIPHERAL_BLOOD: Samples: 302, Genes: 9667
MICROARRAYS|RA|WHOLE_BLOOD: Samples: 513, Genes: 13774
MICROARRAYS|RA|SYNOVIAL_MEMBRANE: Samples: 103, Genes: 13895
MICROARRAYS|T1D|PERIPHERAL_BLOOD: Samples: 79, Genes: 4977
MICROARRAYS|SSc|PERIPHERAL_BLOOD: Samples: 14, Genes: 11853
MICROARRAYS|SSc|WHOLE_BLOOD: Samples: 37, Genes: 14761
MICROARRAYS|SSc|SKIN: Samples: 33, Genes: 10477
MICROARRAYS|SLE|PERIPHERAL_BLOOD: Samples: 812, Genes: 6521
MICROARRAYS|SLE|WHOLE_BLOOD: Samples: 1094, Genes: 8444
MICROARRAYS|SjS|PERIPHERAL_BLOOD: Samples: 222, Genes: 12987
MICROARRAYS|SjS|WHOLE_BLOOD: Samples: 108, Genes: 9683
MICROARRAYS|SjS|PAROTIC_GLAND: Samples: 35, Genes: 22803
MICROARRAYS|SjS|SALIVARY_GLAND: Samples: 15, Genes: 22803
MICROARRAYS|SjS|SALIVA: Samples: 20, Genes: 20399
RNA_SEQ|RA|WHOLE_BLOOD: Samples: 24, Genes: 13635
RNA_SEQ|RA|SYNOVIAL_MEMBRANE: Samples: 180, Genes: 19366
RNA_SEQ|T1D|PERIPHERAL_BLOOD: Samples: 43, Genes: 7433
RNA_SEQ|T1D|WHOLE_BLOOD: Samples: 54, Genes: